### PR TITLE
WebView2 Process Model CoreWebView2Environment sharing clarifications

### PR DESCRIPTION
**Pull request by david-risney**
Got [a question on stackoverflow ](https://stackoverflow.com/questions/73159981/should-i-reuse-the-webview2-environment-for-multiple-webview-instances/73171374) that isn't quite answered by our process model docs around sharing CoreWebView2Environment objects. Updating the process model doc with that clarification.

**Rendered page section for review:**
https://review.docs.microsoft.com/en-us/microsoft-edge/webview2/concepts/process-model?branch=pr-en-us-2093&tabs=csharp#webview2-runtime-processes-and-the-user-data-folder

### CODE REVIEW

File: microsoft-edge/webview2/concepts/process-model.md [hunk 1]
```diff
@@ -53,6 +53,9 @@ When the first `WebView2` instance is created for a given user data folder, the
...
 
+On the other hand, if you create multiple `CoreWebView2Environment` objects configured the same way including sharing the same user data folder, they will all represent the same user data folder and the same associated collection of processes. Use any of them to create a `WebView2` with the one shared user data folder and associated collection of processes.
+
+If you try to create a `CoreWebView2Environment` using a user data folder which another `CoreWebView2Environment` is already using, and you don't configure the two `CoreWebView2Environment` objects the same, for example if they were created with different `CoreWebView2EnvironmentOptions.Language` values, then the second `CoreWebView2Environment` will fail to create `WebView2`s. This is true for `CoreWebView2Environment` objects that are in the same process or in different processes.
```
Comment (mikehoffms) [inline microsoft-edge/webview2/concepts/process-model.md:58]: @JasonAndrewWriter  - fyi.   also Acrolinx would suggest breaking up this long sentence.  If I tried that, I'd use VS Code as a scratchpad in this local .md file, then paste that rewrite into a Suggestion in this PR UI.
```suggestion
If you try to create a `CoreWebView2Environment` using a user data folder which another `CoreWebView2Environment` is already using, and you don't configure the two `CoreWebView2Environment` objects the same, for example if they were created with different `CoreWebView2EnvironmentOptions.Language` values, then the second `CoreWebView2Environment` will fail to create `WebView2` instances. This is true for `CoreWebView2Environment` objects that are in the same process or in different processes.
```

File: microsoft-edge/webview2/concepts/process-model.md [hunk 2]
```diff
@@ -53,6 +53,9 @@ When the first `WebView2` instance is created for a given user data folder, the
 <!-- TODO: update with profile info -->
 The `CoreWebView2Environment` represents a user data folder and the collection of processes associated with it.  A given renderer process is not associated with a single `CoreWebView2` instance, because the renderer process can serve frames in multiple `CoreWebView2` instances that use the same user data folder, depending on website isolation.  See [Per-frame renderer processes - Site Isolation](https://developers.google.com/web/updates/2018/09/inside-browser-part1#site-isolation).
 
+On the other hand, if you create multiple `CoreWebView2Environment` objects configured the same way including sharing the same user data folder, they will all represent the same user data folder and the same associated collection of processes. Use any of them to create a `WebView2` with the one shared user data folder and associated collection of processes.
```
Comment (mikehoffms) [inline microsoft-edge/webview2/concepts/process-model.md:56]: @JasonAndrewWriter fyi
```suggestion
On the other hand, if you create multiple `CoreWebView2Environment` objects that are configured the same way, including sharing the same user data folder, each `CoreWebView2Environment` object will represent the same user data folder and the same associated collection of processes. Use any of those `CoreWebView2Environment` objects to create a `WebView2` with the one shared user data folder and associated collection of processes.
```

**Review by mikehoffms — CHANGES_REQUESTED** [review]
@david-risney - pls evaluate my Suggestions wording eg "objects"/"instances"

**Review by JasonAndrewWriter — APPROVED** [review]

**Review by JasonAndrewWriter — CHANGES_REQUESTED** [review]
If you create multiple `CoreWebView2Environment` objects that are configured the same way, (including sharing the same user data folder), they will represent the same user data folder and the same associated collection of processes. Use any of these `CoreWebView2Environment` objects to create a `WebView2` with the one shared user data folder and associated collection of processes.

File: microsoft-edge/webview2/concepts/process-model.md [hunk 3]
```diff
@@ -53,6 +53,9 @@ When the first `WebView2` instance is created for a given user data folder, the
 <!-- TODO: update with profile info -->
 The `CoreWebView2Environment` represents a user data folder and the collection of processes associated with it.  A given renderer process is not associated with a single `CoreWebView2` instance, because the renderer process can serve frames in multiple `CoreWebView2` instances that use the same user data folder, depending on website isolation.  See [Per-frame renderer processes - Site Isolation](https://developers.google.com/web/updates/2018/09/inside-browser-part1#site-isolation).
 
+On the other hand, if you create multiple `CoreWebView2Environment` objects that are configured the same way, including sharing the same user data folder, each of these `CoreWebView2Environment` objects will represent the same user data folder and the same associated collection of processes. Use any of these `CoreWebView2Environment` objects to create a `WebView2` with the one shared user data folder and associated collection of processes.
```
Comment (david-risney) [inline microsoft-edge/webview2/concepts/process-model.md:56]:
```suggestion
If you create multiple `CoreWebView2Environment` objects that are configured the same way, (including sharing the same user data folder), they will represent the same user data folder and the same associated collection of processes. Use any of these `CoreWebView2Environment` objects to create a `CoreWebView2` with the one shared user data folder and associated collection of processes.
```

**Comment by david-risney** [comment]
> If you create multiple `CoreWebView2Environment` objects that are configured the same way, (including sharing the same user data folder), they will represent the same user data folder and the same associated collection of processes. Use any of these `CoreWebView2Environment` objects to create a `WebView2` with the one shared user data folder and associated collection of processes.

I've applied these changes. Thanks!

File: microsoft-edge/webview2/concepts/process-model.md [hunk 4]
```diff
@@ -53,6 +53,9 @@ When the first `WebView2` instance is created for a given user data folder, the
 <!-- TODO: update with profile info -->
 The `CoreWebView2Environment` represents a user data folder and the collection of processes associated with it.  A given renderer process is not associated with a single `CoreWebView2` instance, because the renderer process can serve frames in multiple `CoreWebView2` instances that use the same user data folder, depending on website isolation.  See [Per-frame renderer processes - Site Isolation](https://developers.google.com/web/updates/2018/09/inside-browser-part1#site-isolation).
 
+If you create multiple `CoreWebView2Environment` objects that are configured the same way, (including sharing the same user data folder), they will represent the same user data folder and the same associated collection of processes. Use any of these `CoreWebView2Environment` objects to create a `CoreWebView2` with the one shared user data folder and associated collection of processes.
```
Comment (mikehoffms) [inline microsoft-edge/webview2/concepts/process-model.md:56]:
```suggestion
If you create multiple `CoreWebView2Environment` objects that are configured the same way (including sharing the same user data folder), they will represent the same user data folder and the same associated collection of processes. Use any of these `CoreWebView2Environment` objects to create a `CoreWebView2` with the one shared user data folder and associated collection of processes.
```

**Review by mikehoffms — APPROVED** [review]

**Review by JasonAndrewWriter — APPROVED** [review]

File: microsoft-edge/webview2/concepts/process-model.md [hunk 5]
```diff
@@ -53,6 +53,9 @@ When the first `WebView2` instance is created for a given user data folder, the
 <!-- TODO: update with profile info -->
 The `CoreWebView2Environment` represents a user data folder and the collection of processes associated with it.  A given renderer process is not associated with a single `CoreWebView2` instance, because the renderer process can serve frames in multiple `CoreWebView2` instances that use the same user data folder, depending on website isolation.  See [Per-frame renderer processes - Site Isolation](https://developers.google.com/web/updates/2018/09/inside-browser-part1#site-isolation).
```
Comment (mikehoffms) [inline microsoft-edge/webview2/concepts/process-model.md:55]:
```suggestion

### Multiple environment objects

```

Comment (mikehoffms) [reply microsoft-edge/webview2/concepts/process-model.md:55]: Can we use h3 headings to label/group some of these 5 paragraphs?

**Review by jasonstephen15 — APPROVED** [review]

**Pull request merged**